umh-core/pkg/service/redpanda_monitor/redpanda_monitor.go

```diff
@@ -0,0 +1,618 @@
+	}
+
+	// Throughput metrics
+	if family, ok := mf["redpanda_kafka_request_bytes_total"]; ok {
+		// Process only produce/consume metrics in a single pass
+		for _, metric := range family.Metric {
+			if label := getLabel(metric, "redpanda_request"); label != "" {
+				if label == "produce" {
+					metrics.Throughput.BytesIn = getMetricValue(metric)
+				} else if label == "consume" {
+					metrics.Throughput.BytesOut = getMetricValue(metric)
+				}
+			}
+		}
+	}
+
+	// Topic metrics
+	if family, ok := mf["redpanda_kafka_partitions"]; ok {
+		for _, metric := range family.Metric {
+			if topic := getLabel(metric, "redpanda_topic"); topic != "" {
+				metrics.Topic.TopicPartitionMap[topic] = getMetricValue(metric)
+			}
+		}
+	}
+
+	return metrics, nil
+}
+
+// getMetricValue extracts numeric value from a metric
+func getMetricValue(m *dto.Metric) int64 {
+	if m.Counter != nil {
+		return int64(m.Counter.GetValue())
+	}
+	if m.Gauge != nil {
+		return int64(m.Gauge.GetValue())
+	}
+	if m.Untyped != nil {
+		return int64(m.Untyped.GetValue())
+	}
+	return 0
+}
+
+// getLabel extracts a label value from a metric
+func getLabel(m *dto.Metric, name string) string {
+	for _, label := range m.Label {
+		if label.GetName() == name {
+			return label.GetValue()
+		}
+	}
+	return ""
+}
+
+// Status checks the status of a redpanda service
+func (s *RedpandaMonitorService) Status(ctx context.Context, filesystemService filesystem.Service, tick uint64) (ServiceInfo, error) {
+	if ctx.Err() != nil {
+		return ServiceInfo{}, ctx.Err()
+	}
+
+	s6ServiceName := s.getS6ServiceName()
+
+	// First, check if the service exists in the S6 manager
+	// This is a crucial check that prevents "instance not found" errors
+	// during reconciliation when a service is being created or removed
+	if _, exists := s.s6Manager.GetInstance(s6ServiceName); !exists {
+		return ServiceInfo{}, ErrServiceNotExist
+	}
+
+	// Get S6 state
+	s6StateRaw, err := s.s6Manager.GetLastObservedState(s6ServiceName)
+	if err != nil {
+		if strings.Contains(err.Error(), "instance "+s6ServiceName+" not found") ||
+			strings.Contains(err.Error(), "not found") {
+			return ServiceInfo{}, ErrServiceNotExist
+		}
+	}
+
+	s6State, ok := s6StateRaw.(s6fsm.S6ObservedState)
+	if !ok {
+		return ServiceInfo{}, fmt.Errorf("observed state is not a S6ObservedState: %v", s6StateRaw)
+	}
+
+	// Get FSM state
+	fsmState, err := s.s6Manager.GetCurrentFSMState(s6ServiceName)
+	if err != nil {
+		if strings.Contains(err.Error(), "instance "+s6ServiceName+" not found") ||
+			strings.Contains(err.Error(), "not found") {
+			return ServiceInfo{}, ErrServiceNotExist
+		}
+		return ServiceInfo{}, fmt.Errorf("failed to get current FSM state: %w", err)
+	}
+
+	// Get logs
+	s6ServicePath := filepath.Join(constants.S6BaseDir, s6ServiceName)
+	logs, err := s.s6Service.GetLogs(ctx, s6ServicePath, filesystemService)
+	if err != nil {
+		return ServiceInfo{}, fmt.Errorf("failed to get logs: %w", err)
+	}
+
+	// Get metrics
```

Comment:
comment doesnt match, "parseRedpandaLogs" and then put it into metrics variable?